src/Common/OpenSSLHelpers.cpp

```diff
@@ -26,10 +31,15 @@ void encodeSHA256(std::string_view text, unsigned char * out)
 }
 void encodeSHA256(const void * text, size_t size, unsigned char * out)
 {
-    SHA256_CTX ctx;
-    SHA256_Init(&ctx);
-    SHA256_Update(&ctx, reinterpret_cast<const UInt8 *>(text), size);
-    SHA256_Final(out, &ctx);
+    EVP_MD_CTX* ctx = EVP_MD_CTX_new();
+    SCOPE_EXIT(EVP_MD_CTX_free(ctx));  // Or use RAII
+
+    if (!EVP_DigestInit_ex(ctx, EVP_sha256(), nullptr) ||
```

Comment:
This would be a bit prettier:

```cpp
if (!EVP_DigestInit_ex(ctx, EVP_sha256(), nullptr)
    throw Exception(ErrorCodes::OPENSSL_ERROR, "SHA256 DigestInit failed: {}", getOpenSSLErrors());

if (! [...])
    [...]
```